src/backend/libpq/rangerrest.c

```diff
@@ -49,10 +49,26 @@ char* AclObjectKindStr[] =
 	"none"               /* MUST BE LAST */
 };
 
+static int request_id = 1;
+
+void getClientIP(char *remote_host)
+{
+	Port *port = MyProcPort;
+	char remote_port[32];
+	remote_port[0] = '\0';
+
+	int ret = pg_getnameinfo_all(&port->raddr.addr, port->raddr.salen,
+					remote_host, sizeof(remote_host),
+					remote_port,	 sizeof(remote_port),
+					NI_NUMERICHOST | NI_NUMERICSERV);
+	if (ret){
+		elog(LOG,"cannot get clientIP. pg_getnameinfo_all() failed: %s", gai_strerror(ret));
+	}
+	elog(DEBUG3, "get clientIP when building json request : %s", remote_host);
+}
 RangerACLResult parse_ranger_response(char* buffer)
 {
-	Assert(buffer != NULL);
-	if (strlen(buffer) == 0)
+	if (buffer == NULL || strlen(buffer) == 0)
```

Comment:
Thanks @stanlyxiang
If RPS reponse is empty and not --enable-cassert, will cause hawq process to core dump in *parse_ranger_response()*.